src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -165,6 +160,29 @@ internal sealed class ImmutabilityContext {
 			}
 		}
 
+		private bool TryGetImmutableTypeInfo(

```

Comment:
Slight difference, in that `m_extraImmutableTypes` isn't considered if a type already has `[Immutable*]` attributes